plugins/schema/src/main/java/com/github/bsideup/liiklus/schema/JsonSchemaPreProcessor.java

```diff
@@ -88,7 +90,23 @@ public JsonSchemaPreProcessor(URL schemaURL, JsonPointer eventTypePointer, boole
                     );
                 }
 
-                event = (ObjectNode) JSON_MAPPER.readTree(cloudEvent.getData().orElse(null));
+                Object data = cloudEvent.getData().orElseThrow(() -> {
+                    return new IllegalArgumentException("data is missing");
+                });
+
+                if (data instanceof JsonNode) {
```

Comment:
FYI in CloudEvents SDK v2 there will be a generic way of getting `byte[]` (or `ByteBuffer`) from any `CloudEvent` to avoid these `instanceof` checks